src/http_client/request.rs

```diff
@@ -0,0 +1,83 @@
+use std::{fmt, future::Future, pin::Pin};
+
+use anyhow::{anyhow, Error};
+use futures_util::TryFutureExt;
+use http::{request::Builder, HeaderName, HeaderValue};
+use hyper::{body::Bytes, Body, Response};
+
+use super::client::Client;
+
+pub struct RequestBuilder {
```

Comment:
use https://crates.io/crates/derive_builder,  it's been used in the `fluvio` successfully